### ISSUE TITLE
(everything) Conflicts with Everything game from steam

### ISSUE THREAD

**Issue by hugobuddel** [chocolatey-community/chocolatey-coreteampackages#1168]
I've got both the [steam game Everything](https://store.steampowered.com/app/582270/Everything/) and the [search app Everything](https://chocolatey.org/packages/Everything) installed. IIRC, the game was installed first, but not sure. Chocolatey detects the steam game as the location of the Everything search app when doing a `choco upgrade`. 

It seems to install the search app in the proper directory ("C:\Program Files\Everything"), so not in the game directory (which I originally thought). It does however start the game at the end of the upgrade, with Steam warning about the unusual parameters that the install script passes.

This issue is created [on request by an admin on disqus](https://chocolatey.org/packages/Everything#comment-4244643738).

I can do some testing if necessary. I've not yet used either the game or the app much, so I'm not sure whether they influence each other. I'm also willing buy one of the developers a copy of the game for testing :yum:.

## Expected Behavior
The Everything app should be independent from the Everything game instead of the app.

It seems that [this line](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/automatic/everything/tools/chocolateyInstall.ps1#L17) detects the location of the game:

    $installLocation = Get-AppInstallLocation $packageName

## Current Behavior
Output of `choco upgrade`:

```
You have everything v1.4.1895.20180530 installed. Version 1.4.1922 is available based on your source(s).
Progress: Downloading Everything 1.4.1922...
Everything v1.4.1922
everything package files upgrade completed. Performing other installation steps.
Installing 64-bit Everything...
Everything has been installed. 
Everything installed to 'C:\Program Files (x86)\Steam\steamapps\common\Everything'
Everything registered as Everything
Post install command line: . 'C:\Program Files (x86)\Steam\steamapps\common\Everything\Everything.exe' --disable-run-as-admin --install-service
Starting Everything
everything may be able to be automatically uninstalled.
ShimGen has successfully created a shim for es.exe 
The upgrade of everything was successful.
Software installed as 'exe', install location is likely default.
```

I'm not sure what the output was when originally installing the Everything app.

## Possible Solution
Don't know, I don't know much about windows scripting / powershell. Perhaps it would be possible to detect the game and never automatically install in the default steam folder?

## Steps to Reproduce (for bugs)
1. Install Everything the game
2. Install Everything the search app

Perhaps the problem only happens when upgrading the search app, not sure.

## Context
It seems to not cause any problems with either the search app or the game. So maybe there is no real problem. But I'm not that familiar with either to be confident enough to tell the difference.

## Your Environment
* Package Version used: 1.4.1922
* Operating System and version: Windows 10 17134.471
* Chocolatey version: v0.10.11
* Install/uninstall gist: pasted above

**Comment by majkinetor** [comment]
Hah, nice catch and game info :) I will see what I can do about this but probably chocolatey-core module needs to take this into account.

**Comment by majkinetor** [comment]
> I'm also willing buy one of the developers a copy of the game for testing 

I have some game/dev time now, if you are still up for it I can fix it (my steam acc has the same name as github).

**Comment by hugobuddel** [comment]
For sure @majkinetor. I could not find you on Steam. I think the account name and profile name are separate on Steam.

So I've bought the game as a gift on the HumbleBundle store and had it sent to the email address you list on your github profile. Claiming the gift should get you a steam key that you can enter in Steam.

**Comment by majkinetor** [comment]
Got it. Thanks. Will try to fix the issue in next few days.

**Comment by majkinetor** [comment]
@hugobuddel

I reproduced the problem after installing the game (sadly, can't play it ATM, lags too much on my current machine). The problem is usage of `Get-AppInstallLocation` which calls `Get-UninstallRegistryKey` before checking a file system. This function exists after first match so it can't find subsequent match. This is not a bug.

Solutions include:

* not using  `Get-AppInstallLocation` 
* game the system

I decided to game the system via problem that is evident in #784 - `Get-UninstallRegistryKey` uses globs while file system search uses Regular Expressions. Hence I changed the code to use `everything$` as a package name which will in effect skip the registry search and provide the path via file system search.

I think this is OK as I don't believe its possible to fix #784 and it probably requires creation of alternative function and deprecation of current one, so this will hold the water in the future.

Please check it out, this is already published. I tested this on my system and it works.

**Comment by hugobuddel** [comment]
This indeed seems to work fine. Thank you @majkinetor! Also thanks for the explanation, this is an interesting workaround :-). I'll close the issue.
